pandas/_libs/lib.pyx

```diff
@@ -1511,6 +1517,21 @@ cpdef bint is_integer_array(ndarray values):
     return validator.validate(values)
 
 
+cdef class IntegerNaValidator(Validator):
+    cdef inline bint is_value_typed(self, object value) except -1:
+        return util.is_integer_object(value) or util.is_nan(value)
+
+    cdef inline bint is_array_typed(self) except -1:
```

Comment:
I am not sure you need this one, is it actually hit?

Reply:
Removed